torch/csrc/generic/serialization.cpp

```diff
@@ -22,7 +22,13 @@ void THPStorage_(writeFileRaw)(THWStorage *self, io fd)
   data = (scalar_t*)cpu_data.get();
   THCudaCheck(cudaMemcpy(data, THWStorage_(data)(LIBRARY_STATE self), size * sizeof(scalar_t), cudaMemcpyDeviceToHost));
 #endif
-  doWrite(fd, &size, sizeof(int64_t));
+  if (THP_nativeByteOrder() == THPByteOrder::THP_LITTLE_ENDIAN)
+    doWrite(fd, &size, sizeof(int64_t));
+  else {
+    int64_t nsize = size; // convert big endian cpu to little endian storage
```

Comment:
Nit: My preference would be not to initialize `nsize` here or below, so that Asan will correctly detect erroneous usage if it ever arises.

Reply:
@resistor I agree with the nit.  I made the change you requested in commit https://github.com/abalib/pytorch/commit/4dcc2a20feb0122fa1a81a7ad685a3e151505f56